### ISSUE TITLE
v3.1.0-beta.7: Kein Umbruch oder text-overflow: ellipsis im "HomeKitTile" möglich

### ISSUE THREAD

**Issue by somansch** [Zefau/ioBroker.jarvis#2012]
Für die Darstellung des aktuellen Titels der Sonos verwende ich HomeKitTile. Leider sind manchmal die Titelnnamen zu lang, somit wäre ein Zeilenumbruch oder -abbruch mit (...) sinnvoll. Dies funktioniert auch bei PopUps, jedoch nicht im "HomeKitTile". Das setzen von `"text-overflow": "ellipsis !important"` im Datenpunktstil hat keine Auswirkung. Ein generelles `text-align:left` wäre auch sinnvoll.

![image](https://user-images.githubusercontent.com/47367267/198893841-dfea3a48-6b2d-44f6-9713-40e1f4ccf6ba.png)

**Comment by mcuiobroker** [comment]
Der Text bekommt noch extra ein Leerzeichen vorne und hinten?
![image](https://user-images.githubusercontent.com/84818934/198896562-c696c813-a23e-4831-a6b1-96ace9cf2bf3.png)

**Comment by somansch** [comment]
Ergänzung: Der Umbruch funktioniert mit `"white-space": "pre-line !important",`.

Jedoch funktioniert der text-overflow nicht, auch nicht im PopUp. Getestet mit folgenden Einstellungen im DP Stil:
`    "white-space": "nowrap !important",
    "overflow": "hidden !important",
    "text-overflow": "ellipsis !important",`

**Comment by somansch** [comment]
Das Ziel sollte sein, dass alle Widgets standardmäßig den Zeilenumbruch aktiviert haben. Zusätzlich sollte es möglich sein, **pro Widget** das Abschneiden des Inhalts (text-overflow: ellipsis) im UI einstellen zu können, oder zumindest per CSS definieren zu können.

**Comment by somansch** [comment]
@Zefau , gibt es inzwischen eine Lösung für "text-overflow"?

**Comment by Zefau** [comment]
Siehe Vorpost https://github.com/Zefau/ioBroker.jarvis/issues/2012#issuecomment-1301180624

**Comment by mcuiobroker** [comment]
Scheint gar nicht zu funktionieren?
```
.jarvis-device-85395f90-68d8-4c79-82f3-bd9479438b2a .q-textarea .InputAction {
	text-overflow: ellipsis !important;
}
```

![image](https://user-images.githubusercontent.com/84818934/203057929-6168e791-ae62-4c4a-800a-f7b1a228a3d9.png)

**Comment by Zefau** [comment]
```
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
```

**Comment by somansch** [comment]
@Zefau , das hatte ich bereits im DP Stil getestet (siehe weiter oben). Gibt es hier eine Änderung in späteren Betas? Bin auf Beta 12.

**Comment by mcuiobroker** [comment]
Beispiel
![image](https://user-images.githubusercontent.com/84818934/203438481-ad17e678-a6f9-40e6-85d6-62ad165b6591.png)
![image](https://user-images.githubusercontent.com/84818934/203438515-1ec683b3-fdb0-4361-8266-d981184adcab.png)

**Comment by mcuiobroker** [comment]
Problem ist das text-overflow nur mit width in px funktioniert. Aber nicht mit width: %.
https://stackoverflow.com/questions/17779293/css-text-overflow-ellipsis-not-working

**Comment by mcuiobroker** [comment]
@somansch WidgetID anpassen:

```
.jarvis-widget-de074d08-7657-48d5-8aff-0f8e3f3c938b .middle .center {
	width: 300px;
}

.jarvis-widget-de074d08-7657-48d5-8aff-0f8e3f3c938b .middle .center  .text-wrap {
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis !important;
}
```

**Comment by somansch** [comment]
@mcuiobroker , @Zefau danke für euren Input. Habe heute ausgiebig getestet. Es funktioniert über CSS nur für "State", jedoch nicht für "Sekundäre States".

**Comment by mcuiobroker** [comment]
Versuch mal das:
```
.jarvis-widget-de074d08-7657-48d5-8aff-0f8e3f3c938b .middle .center {
	width: 200px;
}

.jarvis-widget-de074d08-7657-48d5-8aff-0f8e3f3c938b .middle .center  .text-wrap {
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis !important;
}

.jarvis-widget-de074d08-7657-48d5-8aff-0f8e3f3c938b .bottom .center .jarvis-State {
	width: 200px;
}

.jarvis-widget-de074d08-7657-48d5-8aff-0f8e3f3c938b .bottom .center  .text-wrap {
	white-space: nowrap;
	overflow: hidden;
	text-overflow: ellipsis !important;
}
```

**Comment by somansch** [comment]
@mcuiobroker , mit der zusätzlichen Angabe einer festen Breite geht es. Jedoch macht das für das responsive design dann leider keinen Sinn. 

Ich habe als "Workaround" jetzt "State" und "Sekundärer State" vertauscht.

**Comment by mcuiobroker** [comment]
Können wir das dann schließen?

**Comment by somansch** [comment]
Ja, aber bei dir wird auch nicht "bottom center" auch nicht gekürzt. Funktioniert so auch nicht. Ich kann es schliessen, jedoch gibt es für die "Sekundären States" keine Lösung.auch nicht

**Comment by mcuiobroker** [comment]
![jarvis-v310-beta36 Scripts css variable setzen](https://user-images.githubusercontent.com/84818934/203876677-08125962-6529-4220-92f0-c0cd6377697b.gif)

**Comment by Zefau** [comment]
Was machen wir mit diesem Issue?

**Comment by mcuiobroker** [comment]
Aus meiner Sicht hat es keine Priorität? Würde es offen lassen, aber zurückstellen.
